src/utils/tree-count.ts

```diff
@@ -0,0 +1,106 @@
+import { SearchMessagesArguments } from "@slack/web-api";
+import format from "date-fns/format";
+import startOfMonth from "date-fns/startOfMonth";
+import endOfMonth from "date-fns/endOfMonth";
+import app from "./slack-app";
+import { SLACK_BOT_USER_ID, TREE_EMOJI_NAME } from "./constants";
+import logger from "./logger";
+import { Match } from "@slack/web-api/dist/response/SearchMessagesResponse";
+
+export const getAllMessages = async (query: string) => {
+  let currentPage = 1;
+
+  const searchConfig: SearchMessagesArguments = {
+    query,
+    token: process.env.SLACK_USER_TOKEN,
+    sort: "timestamp",
+    sort_dir: "asc",
+  };
+
+  const allMessages = [];
+
+  // Get the first round of messages and find out how many pages of messages we have
+  const initialQuery = await app.client.search.messages({
+    ...searchConfig,
+    page: currentPage,
+  });
+
+  // Add initial messages to the array
+  const { messages } = initialQuery;
+  if (messages?.matches?.length) {
+    allMessages.push(...messages.matches);
+  }
+
+  // If we have more than one page of messages, loop through the rest of the pages and grab the messages
+  if (messages?.paging?.pages) {
+    while (messages.paging.pages > currentPage) {
+      currentPage += 1;
+
+      const res = await app.client.search.messages({
+        ...searchConfig,
+        page: currentPage,
+      });
+
+      if (res.messages?.matches?.length) {
+        allMessages.push(...res.messages.matches);
+      }
+    }
+  }
+
+  return allMessages;
+};
+
+// Get messages where the bot has reacted with a 🌳 emoji
+const getReaction = async (message: Match) => {
+  if (message.ts && message.channel) {
+    const reaction = await app.client.reactions.get({
+      token: process.env.SLACK_BOT_TOKEN,
+      channel: message.channel.id,
+      timestamp: message.ts,
+    });
+
+    if (reaction.message?.reactions?.length && SLACK_BOT_USER_ID) {
+      const { users, name } = reaction.message.reactions[0];
```

Comment:
Is this assuming that the bot's reaction will be the first reaction?

Reply:
Yes it is 😄🤦